### PR TITLE
Do not create any producer if the output type of a function is void

### PR DESCRIPTION
**Pull request by srkukarni**
### Motivation

If the function output type is void and an output topic is specified, currently we create a producer thats wasteful(since nothing is going to be produced anyways). This pr avoids the creation in that scenario

### Modifications

Describe the modifications you've done.

### Result

After your change, what will change.

### CODE REVIEW

**Comment by jerrypeng** [comment]
@srkukarni I think you also need to check if the function result is null, don't call write on the sink

**Comment by srkukarni** [comment]
@jerrypeng added the check during write.

File: pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/sink/PulsarSink.java [hunk 1]
```diff
@@ -235,6 +239,7 @@ public void open(Map<String, Object> config, SinkContext sinkContext) throws Exc
 
     @Override
     public void write(Record<T> record) throws Exception {
+        if (record == null) return;
```
Comment (jerrypeng) [inline pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/sink/PulsarSink.java]: Actually I don't think this is needed:
https://github.com/apache/pulsar/blob/master/pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java#L318

Comment (srkukarni) [reply pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/sink/PulsarSink.java]: I think the JavaInstanceRunnable is probably a better check.

Comment (jerrypeng) [reply pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/sink/PulsarSink.java]: Yup that is what I mean.  We don't need the check in  sink "write"

**Comment by srkukarni** [comment]
@jerrypeng reverted the check in write

**Comment by srkukarni** [comment]
run integration tests

**Pull request merged**